src/coreclr/tools/ILVerification/ILImporter.Verify.cs

```diff
@@ -2160,7 +2161,7 @@ void ImportAddressOfField(int token, bool isStatic)
                 instance = null;
 
                 if (field.IsInitOnly)
-                    Check(_method.IsStaticConstructor && field.OwningType == _method.OwningType, VerifierError.InitOnly);
+                    isReadOnly = true;
```

Comment:
As discussed in https://github.com/dotnet/runtime/issues/57444#issuecomment-2617793834, isReadOnly means controlled mutability in the current implementation of IL verify.

If you would like to workaround this without implementing proper implementation of read-only references, should this rather be `                // TODO: verification of readonly references https://github.com/dotnet/runtime/issues/57444
` like for the instance field cases to make it less confusing?

Reply:
Yeah. that's probably great idea, even I plan to subsequently start working on https://github.com/dotnet/runtime/issues/57444 . I still confuse what `isReadonly` mean in the code, is this "new" readonly-references, or this is "old" controlled mutability references. I assume that now this is controlled mutability references have confusing name `isReadonly`/have flag `ReadOnly`

Reply:
> is this "new" readonly-references, or this is "old" controlled mutability references.

It should be the old controlled mutability. The new readonly references did not exist when this code was written.